dotnet/src/SemanticKernel.Abstractions/Services/IAIServiceSelector.cs

```diff
@@ -9,15 +11,26 @@ namespace Microsoft.SemanticKernel;
 public interface IAIServiceSelector
 {
     /// <summary>
-    /// Return the AI service and requesting settings from the specified provider based on the model settings.
-    /// The returned value is a tuple containing instances of <see cref="IAIService"/> and <see cref="PromptExecutionSettings"/>
+    /// Resolves an <see cref="IAIService"/> and associated <see cref="PromptExecutionSettings"/> from the specified
+    /// <see cref="Kernel"/> based on a <see cref="KernelFunction"/> and associated <see cref="KernelArguments"/>.
     /// </summary>
-    /// <typeparam name="T">Type of AI service to return</typeparam>
+    /// <typeparam name="T">
+    /// Specifies the type of the <see cref="IAIService"/> required. This must be the same type
+    /// with which the service was registered in the <see cref="IServiceCollection"/> orvia
+    /// the <see cref="IKernelBuilder"/>.
+    /// </typeparam>
     /// <param name="kernel">The <see cref="Kernel"/> containing services, plugins, and other state for use throughout the operation.</param>
-    /// <param name="function">Semantic Kernel callable function interface</param>
+    /// <param name="function">The function.</param>
     /// <param name="arguments">The function arguments.</param>
-    /// <returns></returns>
+    /// <param name="service">The selected service, or null if none was selected.</param>
+    /// <param name="serviceSettings">The settings associated with the selected service. This may be null even if a service is selected.</param>
+    /// <returns>true if a matching service was selected; otherwise, false.</returns>
 #pragma warning disable CA1716 // Identifiers should not match keywords
-    (T?, PromptExecutionSettings?) SelectAIService<T>(Kernel kernel, KernelFunction function, KernelArguments arguments) where T : class, IAIService;
+    bool TrySelectAIService<T>(
```

Comment:
With this change, the introduced `try/catch` block from KernelFunctionFromPrompt` can be removed in favor of this try check.